tests/unit/openstack/test_environment.py

```diff
@@ -110,6 +113,51 @@ def test_get_neutron_driver_existing(self, mock_neutron_driver_init, mock_keysto
+        mock_auth.build_os_auth.return_value = mock_os_auth
+        mock_keystone_session = mock_keystone_session_init.return_value
+        certs_dir = os.path.dirname(os.path.abspath(certs.__file__))
+        ca_cert_path = os.path.join(certs_dir, 'ca.cert')
+        with open(ca_cert_path, 'r') as f:
+            ca_cert = f.read()
+        client_cert_path = os.path.join(certs_dir, 'client.cert')
+        with open(client_cert_path, 'r') as f:
+            client_cert = f.read()
+        client_key_path = os.path.join(certs_dir, 'client.key')
+        with open(client_key_path, 'r') as f:
+            client_key = f.read()
+        location = OpenstackDeploymentLocation('testdl', 'http://testip', mock_auth, ca_cert=ca_cert, client_cert=client_cert, client_key=client_key)
+        try:
+            created_session = location.create_session()
+            mock_keystone_session_init.assert_called_once_with(auth=mock_os_auth, verify=location._OpenstackDeploymentLocation__ca_cert_path, cert=(location._OpenstackDeploymentLocation__client_cert_path, location._OpenstackDeploymentLocation__client_key_path))
+            self.assertTrue(os.path.exists(location._OpenstackDeploymentLocation__ca_cert_path))
+            self.assertTrue(os.path.exists(location._OpenstackDeploymentLocation__client_cert_path))
+            self.assertTrue(os.path.exists(location._OpenstackDeploymentLocation__client_key_path))
+        finally:
+            location.close()
+
+    @patch('osvimdriver.openstack.environment.keystonesession.Session')
+    def test_close_session_with_certs_removes_files(self, mock_keystone_session_init):
+        mock_os_auth = MagicMock()
+        mock_auth = MagicMock()
+        mock_auth.build_os_auth.return_value = mock_os_auth
+        mock_keystone_session = mock_keystone_session_init.return_value
+        location = OpenstackDeploymentLocation('testdl', 'http://testip', mock_auth, ca_cert='cacert', client_cert='clientcert', client_key='clientkey')
+        try:
+            created_session = location.create_session()
+            mock_keystone_session_init.assert_called_once_with(auth=mock_os_auth, verify=location._OpenstackDeploymentLocation__ca_cert_path, cert=(location._OpenstackDeploymentLocation__client_cert_path, location._OpenstackDeploymentLocation__client_key_path))
+            self.assertTrue(os.path.exists(location._OpenstackDeploymentLocation__ca_cert_path))
+            self.assertTrue(os.path.exists(location._OpenstackDeploymentLocation__client_cert_path))
+            self.assertTrue(os.path.exists(location._OpenstackDeploymentLocation__client_key_path))
+            location.close()
+            self.assertFalse(os.path.exists(location._OpenstackDeploymentLocation__ca_cert_path))
+            self.assertFalse(os.path.exists(location._OpenstackDeploymentLocation__client_cert_path))
+            self.assertFalse(os.path.exists(location._OpenstackDeploymentLocation__client_key_path))
+        finally:
```

Comment:
I guess you don't need the second location.close()?

Reply:
It's there in case the test fails before the other close is called.